### PR TITLE
LPS-81859 Portlet 3.0: Implement new methods in javax.portlet.PortletResponse

### PR DESCRIPTION
**Pull request by ngriffin7a**

### CODE REVIEW

**Comment by ngriffin7a** [comment]
ci:test

**Comment by ngriffin7a** [comment]
ci:reopen

**Comment by ngriffin7a** [comment]
ci:retest

File: portal-impl/src/com/liferay/portlet/internal/PortletResponseImpl.java [hunk 1]
```diff
@@ -446,17 +449,63 @@ public PortletRequestImpl getPortletRequest() {
...
+		Object value = _headers.get(key);
+
+		if (value == null) {
+			return Collections.emptySet();
+		}
+
+		List<String> values = new ArrayList<>();
+
+		if (value instanceof Object[]) {
+			Object[] valueArray = (Object[])value;
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portlet/internal/PortletResponseImpl.java]: @ngriffin7a Isn't this breaking the api contract? The api says 

> This method considers only response properties set or added via setProperty(java.lang.String, java.lang.String) or addProperty(java.lang.String, java.lang.String)

All other headers will end up being an Object[] and passing the if check, then getting converted to string for returning.

I think you meant to do "instanceof String[]", right?

Comment (ngriffin7a) [reply portal-impl/src/com/liferay/portlet/internal/PortletResponseImpl.java]: @shuyangzhou Good catch. Would you like me to make the change and send a new PR?

Comment (shuyangzhou) [reply portal-impl/src/com/liferay/portlet/internal/PortletResponseImpl.java]: Yes, please

**Comment by ngriffin7a** [comment]
See https://github.com/shuyangzhou/liferay-portal/pull/6439

**Pull request closed without merging**